smoothdata.py

```diff
@@ -0,0 +1,52 @@
+import scipp as sc
+
+
+def smoothdata(variable, dim=None, NPoints=3):
+    """
+    Function that smooths data by assigning the value of each point to the
+    mean of the values from surrounding points and itself. The number of points
+    to average is NPoint, which is odd so that the point itself is in the
+    center of the averaged interval. If an even NPoints is given, it is
+    incremented. At the ends of the interval the full number of points is not
+    used, but all available within NPoints//2 is.
+
+    Parameters
+    ----------
+        variable: scipp variable
+            The variable which should have its values smoothed
+
+        dim: scipp Dim
+            The dimension along which values should be smoothed
+
+        NPoints: int
+            The number of points to use in the mean (odd number)
+    """
+
+    if dim is None:
+        raise ValueError("smoothdata was not given a dim to smooth.")
+
+    if NPoints < 3:
+        raise ValueError("smoothdata needs NPoints of 3 or higher.")
+
+    data_length = len(variable[dim, :].values)
```

Comment:
I am not sure what the intention here is? Are you trying to get the length in dimensions `dim`? What you are actually getting is the length of the outermost dimension (which should be what you want if the input is 1-D).

Is this what you are looking for:
```python
data_length = dict(zip(variable.dims, variable.shape))[dim]
```
This is a bit cumbersome, we should consider adding a helper to scipp that does this.

Reply:
I was trying to make sure it would be possible to smooth over one given dim even if several was present. Your solution certainly does what I intended. Will update.